src/py21cmfast/wrapper.py

```diff
@@ -305,25 +312,16 @@ def _setup_inputs(
             ],
         )
 
-    # This turns params into a dict with all the input parameters in it.
-    params = dict(zip(input_params.keys(), params))
-
-    params["user_params"] = UserParams(params["user_params"])
-    params["cosmo_params"] = CosmoParams(params["cosmo_params"])
-
-    if "flag_options" in params:
-        params["flag_options"] = FlagOptions(params["flag_options"])
+    p = convert_input_dicts(*params[:4], defaults=True)
```

Comment:
This might be a misunderstanding on my part, but where have the keys gone here? Can `convert_input_dicts` take a list of values?

Reply:
Yes -- it can either take the four param structs in the order (user, cosmo, astro, flags) or can take named parameters.

Reply:
Ok, but if I understand the output of `_configure_inptuts` correctly then `params[:4]` is not a list of parameter structures, but a list of lists of the values in each struct, without keys. Is it relying on some default ordering?